File: articles/persona-and-social-adaptation-in-analytical-psychology/c/persona_role_scores.c
/*
Persona and Social Adaptation in Analytical Psychology
C example: simple synthetic persona-role score

Synthetic educational demonstration only.
Not for diagnosis, therapy, employment screening, reputation scoring,
social scoring, surveillance, or individual evaluation.
*/

#include <stdio.h>
#include <math.h>

double persona_strength(
    double role_demand,
    double audience_reward,
    double institutional_reward,
    double inward_complexity,
    double reflective_flexibility
) {
    return 0.64 * role_demand
         + 0.48 * audience_reward
         + 0.36 * institutional_reward
         - 0.30 * inward_complexity
         + 0.42 * reflective_flexibility;
}

double persona_rigidity(
    double persona_strength_value,
    double status_dependence,
    double institutional_reward,
    double reflective_flexibility
) {
    return 0.52 * persona_strength_value
         + 0.42 * status_dependence
         + 0.34 * institutional_reward
         - 0.46 * reflective_flexibility;
}

double psychic_strain(
    double persona_strength_value,
    double persona_inward_gap,
    double persona_rigidity_value,
    double shadow_pressure,
    double reflective_flexibility
) {
    return 0.50 * persona_strength_value
         + 0.62 * persona_inward_gap * persona_inward_gap
         + 0.42 * persona_rigidity_value
         + 0.36 * shadow_pressure
         - 0.52 * reflective_flexibility;
}

int main(void) {
    double strength = persona_strength(0.66, 0.50, 0.60, 0.70, 0.76);
    double rigidity = persona_rigidity(strength, 0.34, 0.60, 0.76);
    double strain = psychic_strain(strength, 0.08, rigidity, 0.32, 0.76);

    printf("Synthetic persona strength: %.3f\n", strength);
    printf("Synthetic persona rigidity: %.3f\n", rigidity);
    printf("Synthetic psychic strain: %.3f\n", strain);
    return 0;
}

File: articles/persona-and-social-adaptation-in-analytical-psychology/fortran/persona_role_scores.f90
! Persona and Social Adaptation in Analytical Psychology
! Fortran example: simple synthetic persona-role score

program persona_role_scores_demo
  implicit none

  real :: strength
  real :: rigidity
  real :: strain

  strength = 0.64 * 0.66 + 0.48 * 0.50 + 0.36 * 0.60 - 0.30 * 0.70 + 0.42 * 0.76
  rigidity = 0.52 * strength + 0.42 * 0.34 + 0.34 * 0.60 - 0.46 * 0.76
  strain = 0.50 * strength + 0.62 * 0.08 * 0.08 + 0.42 * rigidity + 0.36 * 0.32 - 0.52 * 0.76

  print *, "Synthetic persona strength:", strength
  print *, "Synthetic persona rigidity:", rigidity
  print *, "Synthetic psychic strain:", strain
end program persona_role_scores_demo
